libs/utils/src/zstd.rs

```diff
@@ -0,0 +1,82 @@
+use std::io::SeekFrom;
+
+use anyhow::{Context, Result};
+use async_compression::{
+    tokio::{bufread::ZstdDecoder, write::ZstdEncoder},
+    zstd::CParameter,
+    Level,
+};
+use camino::Utf8Path;
+use nix::NixPath;
+use tokio::{
+    fs::{File, OpenOptions},
+    io::AsyncBufRead,
+    io::AsyncSeekExt,
+    io::AsyncWriteExt,
+};
+use tokio_tar::{Archive, Builder, HeaderMode};
+use tracing::*;
+use walkdir::WalkDir;
+
+/// Creates a Zstandard tarball.
+pub async fn create_tarball(path: &Utf8Path, tmp: &Utf8Path) -> Result<(File, u64)> {
```

Comment:
I like the old name better tbh. when called via `zstd::create_tarball` it is clear what it is doing, but some people might do `use utils::zstd::create_tarball` and `create_tarball(...)` is not very clear.

Reply:
I would expect people to use it as `zstd::create_tarabll()`, like what is in this PR.

Reply:
whatever, shrug.